### ISSUE TITLE
How do I do a similar example to the one in Figure 3 in the paper of Tang?

### ISSUE THREAD

**Issue by SheperoMah** [davharris/mistnet#115]
I am not a machine learning expert so apologize for the inconvenience. But I have some issues in implementing your code

First the code does not work if you have a y as a one column vector. So I had to create a second column of values equal to one. I believe the code needs that you have multiple output for each training data point.

My Idea is to map from a probability distribution. Assume that you have three values of x, and each value has an output y that follows a different distribution. I tried to do this using Mistnet and I failed. I tried to do several methods: adding layers, changing loss function, changing the distribution of the values of weights, but it never worked.
An intuitive example x is the day of the week, y is the number of visitors of a shop. I have data from the hole year, and each day follows a distribution. I want to predict the number of visitors throughout the next year. Each day of the week will be mapped according to its specific distribution.

Another issue, if you replace the gaussian.prior with a lognormal.prior and change the names for the parameters to meanlog and sdlog the code gives a NaNs estimate.

A pseudocode
[Code.docx](https://github.com/davharris/mistnet/files/860854/Code.docx)

**Comment by micheletufano** [comment]
I am trying to replicate exactly the results of the Tang paper in Figure 3a  (Dataset A).

**Dataset**
The [dataset](https://github.com/davharris/mistnet/files/917499/data.zip) is formed by:
- `X.csv` is the one-dimensional input;
- `Y.csv` is the one-dimensional output;

As @SheperoMah mentioned, it seems that this implementation does not allow one-column vector as input/output. Therefore, I also added another "feature" in the input/output always set at 1.

```
x <- read.csv("data/X.csv");
y <- read.csv("data/Y.csv");
x <- cbind(x, 1);
x <- as.matrix(x);
y <- cbind(y, 1);
y <- as.matrix(y);
```

**Network**
I'm using a mistnet with:
- 4 layers of size 16 with rectify.nonlinearity();
- 1 output layer with sigmoid.nonlinearity();
- loss = squaredLoss()
- n.importance.samples = 30

Attached is the complete [code](https://github.com/davharris/mistnet/files/917505/run_mistnet.R.txt).


**Output**
I plotted the training data (in blue) and the predictions for 3 importance samples (in red).
![dataseta](https://cloud.githubusercontent.com/assets/11789495/24973137/bad8c02c-1f8b-11e7-80b8-6c9af088a529.png)

It doesn't look pretty good.

**Questions**
@SheperoMah were you eventually able to replicate the example you reported above?
@davharris , I'm aware that your implementation is not exactly the Tang's SFNN, but do you think it should still be able to learn from this dataset? Please feel free to tell us if this is completely out of scope for your implementation.

**Comment by SheperoMah** [comment]
No I couldn't.
